File: server/api-module/src/main/java/com/ssafy/api/dto/Message.java
package com.ssafy.api.dto;

import lombok.*;

@Builder
@Setter
@Getter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class Message {

    //입장, 나감, 이야기, 게임시작, 게임종료
    public enum Type {
        ENTER, OUT, TALK, START, END
    }

    private Type type; // 메시지 타입
    private String roomId; // 방번호
    private String sender; // 메시지 보낸사람
    private String message; // 메시지
}

File: aos/Near/app/src/main/java/com/ssafy/near/dto/Message.kt
package com.ssafy.near.dto

import java.io.Serializable

data class Message(var type: MsgType, val roomId: String, val sender: String, val message: String) : Serializable

enum class MsgType {
    ENTER, OUT, TALK, START, END, CHANGE, ANSWER, NOTICE
}
